llmfoundry/data/finetuning/dataloader.py

```diff
@@ -122,8 +123,13 @@ def build_finetuning_dataloader(cfg: DictConfig,
     if tokenizer.pad_token is None:
         tokenizer.pad_token = tokenizer.eos_token
 
+    collate_fn, dataloader_batch_size = _build_collate_fn(
```

Comment:
todo (for another refactor, out of scope for this PR): make `_build_collate_fn` accept not a cfg but arguments directly. This may be hard because if the packing_ratio is "auto", `build_collate_rn` will call `build_finetuning_dataloader` under the hood. Probably will be a bigger refactor than just expanding the arguments of `_build_collate_fn`